### PR TITLE
ENH: implement Timedelta.__mod__ and __divmod__

### PR DESCRIPTION
**Pull request by jbrockmendel**
This implements exclusively the `__mod__, __rmod__, __divmod__, __rdivmod__` parts of #19365, does _not_ implement any of the bugfixes or tests that got rolled into that PR.  Several tests are xfailed for now, will be fixed after #19738.  These tests will be rebased/moved to test_arithmetic after #19752.

- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.23.0.txt [hunk 1]
```diff
@@ -571,6 +585,7 @@ Other API Changes
 - Set operations (union, difference...) on :class:`IntervalIndex` with incompatible index types will now raise a ``TypeError`` rather than a ``ValueError`` (:issue:`19329`)
 - :class:`DateOffset` objects render more simply, e.g. "<DateOffset: days=1>" instead of "<DateOffset: kwds={'days': 1}>" (:issue:`19403`)
 - :func:`pandas.merge` provides a more informative error message when trying to merge on timezone-aware and timezone-naive columns (:issue:`15800`)
+- :func:`Timedelta.__mod__`, :func:`Timedelta.__divmod__` now accept timedelta-like and numeric arguments instead of raising ``TypeError`` (:issue:`19365`)
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.23.0.txt]: This is the same as above? Isn't it enough to mention it once?

File: pandas/tests/scalar/test_timedelta.py [hunk 2]
```diff
@@ -254,6 +254,186 @@ def test_rfloordiv(self):
         with pytest.raises(TypeError):
             ser // td
 
+    def test_td_mod_timedeltalike(self):
```
Comment (jorisvandenbossche) [inline pandas/tests/scalar/test_timedelta.py]: Like `test_rfloordiv` above, I think you can leave out the `_td_` in all test names (the class name already makes clear it is about Timedelta)

**Review by jorisvandenbossche — COMMENTED** [review]
small comment, looks good to me

File: doc/source/whatsnew/v0.23.0.txt (hunk 1 above)
Comment (jbrockmendel) [reply doc/source/whatsnew/v0.23.0.txt]: This is an attempt to follow the instructions in #19365, some of which were unclear.  I'm open to suggestions.

File: pandas/tests/scalar/test_timedelta.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/scalar/test_timedelta.py]: ok

File: doc/source/whatsnew/v0.23.0.txt (hunk 1 above)
Comment (jorisvandenbossche) [reply doc/source/whatsnew/v0.23.0.txt]: My suggestion would be to just remove this one, but I don't know the reason for the previous instruction

Comment (jbrockmendel) [reply doc/source/whatsnew/v0.23.0.txt]: @jrback is there consensus on this suggestion?

File: doc/source/timedeltas.rst [hunk 3]
```diff
@@ -283,6 +283,20 @@ Rounded division (floor-division) of a ``timedelta64[ns]`` Series by a scalar
...
+The mod (%) and divmod operations are defined for ``Timedelta`` when operating with another timedelta-like or with a numeric argument. (:issue:`19365`)
+
+.. ipython:: python
+
+   pd.Timedelta(hours=37) % datetime.timedelta(hours=2)
+
+   # divmod against a timedelta-like returns a pair (int, Timedelta)
+   divmod(datetime.timedelta(hours=2), pd.Timedelta(minutes=11))
+
+   # divmod against a numeric returns a pair (Timedelta, Timedelta)
```
Comment (jreback) [inline doc/source/timedeltas.rst]: nice!

File: doc/source/whatsnew/v0.23.0.txt (hunk 1 above)
Comment (jreback) [reply doc/source/whatsnew/v0.23.0.txt]: oh i guess I wasn't clear. if you make a sub-section then no reason to repeat things, so kill this entry as it dupes the sub-section.

File: pandas/tests/scalar/timedelta/test_arithmetic.py [hunk 4]
```diff
@@ -420,3 +420,189 @@ def test_td_rfloordiv_numeric_series(self):
...
+
+    # ----------------------------------------------------------------
+    # Timedelta.__divmod__, __rdivmod__
+
+    def test_divmod_numeric(self):
+        # GH#19365
+        td = Timedelta(days=2, hours=6)
+
+        result = divmod(td, 53 * 3600 * 1e9)
+        assert result[0] == Timedelta(1, unit='ns')
```
Comment (jreback) [inline pandas/tests/scalar/timedelta/test_arithmetic.py]: slightly OT: we have test_nat.py so either should add this there, or have here, or *maybe* both.

**Review by jreback — CHANGES_REQUESTED** [review]
doc otherwise lgtm (mod suggestion below :)

ping on green.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm ping on green.

**Comment by jbrockmendel** [comment]
Ping

**Comment by jreback** [comment]
thanks!

**Pull request merged**